redbpf/src/lib.rs

```diff
@@ -820,6 +832,43 @@ impl<K: Clone, V: Clone> Iterator for MapIter<'_, '_, K, V> {
     }
 }
 
+impl StackTrace<'_> {
+    pub fn new(map: &Map) -> StackTrace<'_> {
+        StackTrace {
+            base: map
+        }
+    }
+
+    pub fn get(&mut self, id: libc::c_int) -> Option<BpfStackFrames> {
+        unsafe {
+            let mut value = MaybeUninit::uninit();
+
+            let ret = bpf_sys::bpf_lookup_elem(
+                self.base.fd,
+                &id as *const libc::c_int as *mut libc::c_int as _,
+                value.as_mut_ptr() as *mut _
+            );
+
+            if ret == 0 {
+                Some(value.assume_init())
+            } else {
+                None
+            }
+        }
+    }
+
+    pub fn remove(&mut self, id: libc::c_int) -> bool {
+        unsafe {
+            let ret = bpf_sys::bpf_delete_elem(
+                self.base.fd,
+                &id as *const libc::c_int as *mut libc::c_int as _,
+            );
+
+            ret == 0
```

Comment:
This is great! Can you please modify `HashMap::delete` to be consistent with reporting the error from the kernel? I think the best would probably be to return `Result<()>`, and `Err(Error::Map)` for failure in both cases.

Reply:
Change `HashMap::delete` will cause API break, maybe we can add a `HashMap::delete2`?

Reply:
I think it's not worth adding a separate method nor breaking API just for this. I'm planning to overhaul error handling which is now a bit messy soon, so I can fix this when I get to that.